src/backup_observer.py

```diff
@@ -176,10 +187,39 @@ def _on_restore_backup_action(self, event: ActionEvent) -> None:
         container = self._charm.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
 
         try:
-            backup.restore_backup(container, s3_parameters, backup_passphrase, str(backup_id))
+            backup.restore_backup(container, s3_parameters, backup_passphrase, backup_id)
         except (backup.BackupError, APIError, ExecError):
             logger.exception("Error Restoring Backup.")
             event.fail("Error Restoring Backup.")
             return
 
         event.set_results({"result": "correct"})
+
+    def _on_delete_backup_action(self, event: ActionEvent) -> None:
+        """Delete a backup from S3.
+
+        Args:
+            event: Event triggering the delete backup action.
+        """
+        backup_id = event.params["backup-id"]
+        logger.info("backup-id %s is going to be deleted", backup_id)
+
+        try:
+            s3_parameters = backup.S3Parameters(**self._s3_client.get_s3_connection_info())
+        except ValueError:
+            logger.exception("Wrong S3 configuration in delete backup action")
+            event.fail("Wrong S3 configuration in delete backup action. Check S3 integration.")
+            return
+
+        try:
+            s3_client = backup.S3Client(s3_parameters)
+            if not s3_client.exists_backup(backup_id):
+                event.fail("Backup to delete does not exist.")
```

Comment:
I'm wondering if we should fail here or exit with success and still print this output so that the action is idempotent

Reply:
I was thinking in `rm` command that fails if the file does not exist. Open to change it if you still think that this action should behave otherwise...

Reply:
I'm having a similar discussion [her](https://github.com/canonical/digest-squid-auth-helper/pull/2#discussion_r1484034353). So it'd be a good idea to agree on an approach @gregory-schiano

Reply:
I answered on the other side also but same thing: I would issue a warning (or log it) and finish with success

Reply:
Changed so the action does not fail (`status: completed` instead of `status: failed`), but the result message says the backup does not exist and a warning is logged.